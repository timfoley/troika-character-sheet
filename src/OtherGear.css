.wearing {
	grid-area: wearing;
}

.money {
	grid-area: money;
}

.provisions {
	grid-area: provisions;
}

.gear__label {
	font-weight: bold;
	display: flex;
	flex-direction: column;
}

.gear__input {
	height: 100%;
	width: 100%;
	margin: 0;
	border: 1px solid black;
	border-radius: 0;
	font-size: 2rem;
	margin-top: 4px;
}

.wearing textarea {
	font-size: 1rem;
	width: 100%;
}
