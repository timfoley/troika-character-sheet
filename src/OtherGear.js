import React from "react";

import "./OtherGear.css";

export function OtherGear(props) {
	return (
		<React.Fragment>
			<label className="gear__label wearing">
				Wearing
				<textarea className="gear__input" value={props.wearing} />
			</label>

			<label className="gear__label money box">
				Money
				<input
					type="number"
					className="gear__input"
					value={props.money}
				/>
			</label>

			<label className="gear__label provisions box">
				Provisions
				<input
					type="number"
					className="gear__input"
					value={props.provisions}
				/>
			</label>
		</React.Fragment>
	);
}

OtherGear.defaultProps = {
	money: 0,
	provisions: 0,
	wearing: ""
};
