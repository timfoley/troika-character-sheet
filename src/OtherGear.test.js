import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";

import { OtherGear } from "./OtherGear";

it("renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<OtherGear />, div);
	ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
	const { container } = render(<OtherGear />);
	expect(container.firstChild).toMatchSnapshot();
});
