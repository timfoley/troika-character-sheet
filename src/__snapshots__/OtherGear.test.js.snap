// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<label
  class="gear__label wearing"
>
  Wearing
  <textarea
    class="gear__input"
  />
</label>
`;
